api/app/services/knowledge_updates/llm_wiki_update_service.py

```
@@ -1957,6 +1962,76 @@ def _faq_ref_value(source: Dict[str, Any], fallback: str) -> str:
+    try:
+        with sqlite3.connect(db_path) as conn:
+            conn.row_factory = sqlite3.Row
+            table_exists = conn.execute(
+                "SELECT 1 FROM sqlite_master WHERE type = 'table' AND name = 'faqs'"
+            ).fetchone()
+            if table_exists is None:
+                return {}
+
+            placeholders = ",".join("?" for _ in unique_ids)
+            rows = conn.execute(
+                f"SELECT id, question FROM faqs WHERE id IN ({placeholders})",
```

Ruff (0.15.20)

[error] Possible SQL injection vector through string-based query construction

(S608)